### DISCUSSION TITLE
How to upload file by dealing with windows popup file selection in mac os and windows os

### DISCUSSION THREAD

**Discussion by nakodavati** [seleniumbase/SeleniumBase#2512] [General]
Hi 
I am trying to upload a file to my application where it opens os popup when we click on upload file button , i can't find the proper solution by choose_file option, can you let me know .

**Comment by mdmintz** [comment]
Examples of uploading files:
* [examples/upload_file_test.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/upload_file_test.py)
* [examples/test_show_file_choosers.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_show_file_choosers.py)

**Comment by OpsecGuy** [comment]
1 more example from me :)

from an old script of mine. 
```
file_input = self.session.find_element('EXAMPLE_XPATH')
self.session.execute_script("arguments[0].style.display = 'block';", file_input)
file_input.send_keys("C:\\Users\\...\\...\\YOUR_IMAGE.png")
self.session.click("//div[contains(text(),'Accept')]")
self.session.sleep(1.0)
```
